README.rst

```diff
@@ -15,104 +15,171 @@ sqlitedict -- persistent ``dict``, backed-up by SQLite and pickle
-      >>> mydict = SqliteDict('./my_db.sqlite', encode=my_encode, decode=my_decode)
+    with SqliteDict("./db.sqlite") as db:
+        print("There are {} items in the database".format(len(db)))
 
 
-Installation
-------------
+Tables
+------
 
-The module has no dependencies beyond Python itself. The minimum Python version is 2.5, continuously tested on Python 2.6, 2.7, 3.3 and 3.4 `on Travis <https://travis-ci.org/RaRe-Technologies/sqlitedict>`_.
+A default table is used when no table name is specified.
 
-Install or upgrade with::
+.. code-block:: python
 
-    pip install -U sqlitedict
+    from sqlitedict import SqliteDict
 
-or from the `source tar.gz <http://pypi.python.org/pypi/sqlitedict>`_::
+    # support for multiple tables in the same files.
+    # writes are serialized. consider splitting large databases (> 1 GB) for performance.
+    products      =  SqliteDict("./db.sqlite", tablename="product", autocommit=True)
+    manufacturers =  SqliteDict("./db.sqlite", tablename="manufacturer", autocommit=True)
 
-    python setup.py install
+    products["1"] = {"name": "first item",  "manufacturer_id": "1"}
+    products["2"] = {"name": "second item", "manufacturer_id": "1"}
+
+    manufacturers["1"] = {"manufacturer_name": "afactory",          "location": "US"}
+    manufacturers["2"] = {"manufacturer_name": "anotherfactory",    "location": "UK"}
+
+    tables = products.get_tablenames()
+    print(tables)
+    # ["product", "manufacturer"]
 
-Documentation
+Serialization
 -------------
 
-Standard Python document strings are inside the module:
+Keys are strings. Values are any serializeable object.
+
+By default Pickle is used internally to (de)serialize the values.
+
+It's possible to use a custom (de)serializer, notably for JSON and for compression.
 
 .. code-block:: python
 
-  >>> import sqlitedict
-  >>> help(sqlitedict)
+    # use JSON instead of pickle
+    import json
+    mydict = SqliteDict("./my_db.sqlite", encode=json.dumps, decode=json.loads)
+
+    # apply zlib compression after pickling
+    import zlib, pickle, sqlite3
+
+    def my_encode(obj):
+        return sqlite3.Binary(zlib.compress(pickle.dumps(obj, pickle.HIGHEST_PROTOCOL)))
+    def my_decode(obj):
+        return pickle.loads(zlib.decompress(bytes(obj)))
+
+    mydict = SqliteDict("./my_db.sqlite", encode=my_encode, decode=my_decode)
+
+More
+----
 
-(but it's just ``dict`` with a commit, really).
+Functions are well documented, see docstrings directly in ``sqlitedict.py`` or call ``help(sqlitedict)``.
 
 **Beware**: because of Python semantics, ``sqlitedict`` cannot know when a mutable
-SqliteDict-backed entry was modified in RAM. For example, ``mydict.setdefault('new_key', []).append(1)``
-will leave ``mydict['new_key']`` equal to empty list, not ``[1]``. You'll need to
-explicitly assign the mutated object back to SqliteDict to achieve the same effect:
+SqliteDict-backed entry was modified in RAM. For example, ``db["123"]["name"] = "hello world"``
```

Comment:
I think that will raise a KeyError `Key '123' doesn't exist`, not leave `mydict` (what `mydict`? isn't it `db`?) as it was.